src/paths/all_shortest_paths.c

```diff
@@ -293,14 +331,30 @@ int igraph_get_all_shortest_paths(const igraph_t *graph,
         if (geodist[i] > 0) {
             /* yes, copy them to the result vector */
             while (fatherptr != 0) {
-                VECTOR(*res)[j++] = VECTOR(paths)[fatherptr - 1];
+                if (vertices) {
+                    VECTOR(*vertices)[j++] = VECTOR(paths)[fatherptr - 1];
+                } 
+                if (edges) {
+                    VECTOR(*edges)[t++] = VECTOR(path_edge)[fatherptr - 1];
+                }
+                /* If dont return the vertex or the edge path then release*/
+                if (!vertices) {
```

Comment:
Please merge this as an `else` branch with the `if (vertices) {` condition above -- they logically belong together.